__tests__/operation.test.ts

```diff
@@ -933,34 +933,83 @@ describe('#hasRequestBody()', () => {
   });
 });
 
+describe('#getRequestBody()', () => {
+  it('should return false on an operation without a requestBody', async () => {
+    const oas = Oas.init(petstore);
+    await oas.dereference();
+
+    const operation = oas.operation('/pet/findByStatus', 'get');
+    expect(operation.getRequestBody('application/json')).toBe(false);
+  });
+
+  it('should return false on an operation without the specified requestBody media type', async () => {
+    const oas = Oas.init(petstore);
+    await oas.dereference();
+
+    const operation = oas.operation('/pet', 'put');
+    expect(operation.getRequestBody('text/xml')).toBe(false);
+  });
+
+  it('should return the specified requestBody media type', async () => {
+    const oas = Oas.init(petstore);
+    await oas.dereference();
+
+    const operation = oas.operation('/pet', 'put');
+    expect(operation.getRequestBody('application/json')).toStrictEqual({
+      schema: {
+        properties: {
+          category: expect.objectContaining({ 'x-readme-ref-name': 'Category' }),
+          id: expect.any(Object),
+          name: expect.any(Object),
+          photoUrls: expect.any(Object),
+          status: expect.any(Object),
+          tags: {
+            items: expect.objectContaining({ 'x-readme-ref-name': 'Tag' }),
+            type: 'array',
+            xml: {
+              name: 'tag',
+              wrapped: true,
+            },
+          },
+        },
+        required: ['name', 'photoUrls'],
+        type: 'object',
+        xml: {
+          name: 'Pet',
+        },
+        'x-readme-ref-name': 'Pet',
+      },
+    });
+  });
+});
+
 describe('#getResponseByStatusCode()', () => {
   it('should return false if the status code doesnt exist', () => {
     const operation = Oas.init(petstore).operation('/pet/findByStatus', 'get');
     expect(operation.getResponseByStatusCode(202)).toBe(false);
   });
 
-  it('should return the response', () => {
-    const operation = Oas.init(petstore).operation('/pet/findByStatus', 'get');
+  it('should return the response', async () => {
```

Comment:
I had to rework this test a bit because the dereferencing library we use does parameter reassigning so if one test dereferences the petstore spec then tests that use that spec after it will have a dereferenced spec.